### PR TITLE
ImageNet reproducibility: imagenet.jl + README cleanup

### PR DESCRIPTION
**Pull request by vonpetersenn**
## Summary
ImageNet (ResNet50) compression experiments, resulting in reproducible results for DRR, RL1, and PMMP across 90+10 training epochs.

## Infrastructure
- **Julia Package**: restructure Julia codebase as precompilable package
- **Dataset loading**: chunked preprocessed dataloader and
  online JPEG loader for Imagenet-1k; unified API matching the pattern for CIFAR/MNIST
- **Preprocessing**: diverse standard augementations for Imagenet
- **Checkpointing**: checkpoint system with adjective-animal IDs,
  explicit resume by ID, periodic saves

## Experiments
Vanilla baseline: 72.2% top-1 (cosine LR, ρ=8e-6, label smoothing, ep90)

| Method | α | Val acc | CR |
|--------|---|---------|-----|
| DRR | 4e-7 | 73.4% | 87.5% (8.0×) |
| RL1 | 2.5e-6 | 71.1% | 85.2% (6.8×) |
| PMMP | 5e-6 | 70.7% | 71.4% (3.5×) |

## Reproducibility
`imagenet.jl` runs all four configurations above. See README for setup.

### CODE REVIEW

**Review by LUK4S-B — APPROVED** [review]
I am impressed by how much work you have put into this! Thanks for all these contributions!
For next time it would be good to submit more PRs in between to make it easier to review the code.
